CMakeLists.txt

```diff
@@ -137,15 +137,24 @@ if (NOT "${Boost_VERSION}" STRLESS "106100")
     add_library(JIT INTERFACE)
     add_library(VexCL::JIT ALIAS JIT)
 
-    target_link_libraries(JIT INTERFACE
-        Common
-        $<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS} dl>
-        $<$<CXX_COMPILER_ID:Clang>:${OpenMP_CXX_FLAGS} dl>
-        $<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS} dl>
-        )
+    # CMake 3.9, allows correct linking with Clang
+    if(TARGET OpenMP::OpenMP_CXX)
+        target_link_libraries(JIT INTERFACE OpenMP::OpenMP_CXX Common)
+    else()
+        if(CXX_COMPILER_ID STREQUAL Clang)
+            message(WARNING "Clang JIT needs CMake 3.9 to properly detect OpenMP linker flags!")
+        endif()
+        # Clang may need -fopenmp=libiomp5 instead, can't be detected here without CMake 3.9
+        target_link_libraries(JIT INTERFACE
+            $<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS}>
+            $<$<CXX_COMPILER_ID:Clang>:-fopenmp=libomp>
+            $<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS}>
+            )
+        target_compile_options(JIT INTERFACE ${OpenMP_CXX_FLAGS})
+    endif()
```

Comment:
Also, is this change necessary now?

Reply:
Yes, it is better to use the `OpenMP::OpenMP_CXX` target if it exists. I've scaled back some of the changes, though, to keep this clean.

Reply:
On some systems, like Windows as you've seen, linker and compiler flags are different. CMake knows that and handles it appropriately through FindOpenMP (for CMake 3.9+), instead of having to list each one like is done now.